security/tools/sdsclient/main.go

```diff
@@ -0,0 +1,41 @@
+// Program sdsclient simulates a SDS client to test SDS Server, citadel agent.
+package main
+
+import (
+	"log"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/pkg/cmd"
+	"istio.io/istio/security/pkg/testing/sdsc"
+	"istio.io/pkg/env"
+)
+
+var (
+	sdsServerUdsPath = env.RegisterStringVar(
+		"CITADEL_AGENT_TESTING_UDS_PATH", "unix:///var/run/sds/uds_path",
+		"The server unix domain socket path").Get()
+	rootCmd = &cobra.Command{
+		Use: "sdsclient is used for testing sds server",
+		RunE: func(c *cobra.Command, args []string) error {
+			client, err := sdsc.NewClient(sdsc.ClientOptions{
+				ServerAddress: sdsServerUdsPath,
+			})
+			if err != nil {
+				log.Fatalf("failed to create client, error %v", err)
+			}
+			if err := client.Send(); err != nil {
+				log.Fatalf("failed to send sds request err: %v", err)
+			}
+			client.Start()
+			cmd.WaitSignal(make(chan struct{}))
```

Comment:
after get first resp, ```WaitSignal``` return and this func exit, while go routine in ```start``` still running ?

Reply:
I take my comment back, thought it's ```WaitForUpdate``` in sdsclient.go